### PR TITLE
AP_Logger: fixed stuck logging with LOG_DISARMED=1

### PR DESCRIPTION
**Pull request by tridge**
this prevents a stuck write closing logging when LOG_DISARMED=1
this fixes LOG_DISARMED=1 on HWing

### CODE REVIEW

**Comment by peterbarker** [comment]
.... which bit in here is actually required for the hwing?  the 3s -> 5s or the don't-abort-logging bit?

I'm fine with both...

**Comment by tridge** [comment]
> .... which bit in here is actually required for the hwing? the 3s -> 5s or the don't-abort-logging bit?

I think the minimal change is moving the setting of timestamp to top of the function, but I think all the changes are an improvement

File: libraries/AP_Logger/AP_Logger_File.cpp [hunk 1]
```diff
@@ -155,17 +155,9 @@ void AP_Logger_File::periodic_1Hz()
             // we register the IO timer callback
             GCS_SEND_TEXT(MAV_SEVERITY_CRITICAL, "AP_Logger: stuck thread (%s)", last_io_operation);
         }
-        if (io_thread_warning_decimation_counter++ > 57) {
+        if (io_thread_warning_decimation_counter++ > 30) {
```
Comment (andyp1per) [inline libraries/AP_Logger/AP_Logger_File.cpp:158]: Why this change?

**Review by peterbarker — APPROVED** [review]
